File: ui/src/app/models/question.ts
import {User} from "./user";
import {SafeUrl} from "@angular/platform-browser";
import {Tag} from "./tags";

export interface Question {
  qid? : number;
  userId?: User;
  title: string ;
  text: string;
  creationDate: string;
  imagePath:string | null;
  safeUrl? : SafeUrl;
  tags?: Tag[];
}

File: src/main/java/com/stackoverflow/demo/entity/Question.java
package com.stackoverflow.demo.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Entity
@Table(name = "question")
@Data
@NoArgsConstructor
public class Question {
    @Id
    @Column(name = "q_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long qId;

    @ManyToOne()
    @JoinColumn(name = "u_id")
    private User userId;

    @Column(name = "title")
    private String title;

    @Column(name = "creation_date")
    private Timestamp creationDate;

    @Column(name = "text")
    private String text;
    @Column(name = "image_path")
    private String imagePath;
}
